### ISSUE TITLE
flaky test: test_operator.test_dropout

### ISSUE THREAD

**Issue by haojin2** [apache/incubator-mxnet#12314]
http://jenkins.mxnet-ci.amazon-ml.com/blue/organizations/jenkins/incubator-mxnet/detail/PR-12311/1/pipeline
```
======================================================================

FAIL: test_operator.test_dropout

----------------------------------------------------------------------

Traceback (most recent call last):

  File "/usr/local/lib/python3.5/dist-packages/nose/case.py", line 198, in runTest

    self.test(*self.arg)

  File "/work/mxnet/tests/python/unittest/common.py", line 172, in test_new

    orig_test(*args, **kwargs)

  File "/work/mxnet/tests/python/unittest/test_operator.py", line 5844, in test_dropout

    check_dropout_ratio(0.0, shape)

  File "/work/mxnet/tests/python/unittest/test_operator.py", line 5788, in check_dropout_ratio

    assert exe.outputs[0].asnumpy().min() == min_value

AssertionError: 

-------------------- >> begin captured logging << --------------------

common: INFO: Setting test np/mx/python random seeds, use MXNET_TEST_SEED=119185068 to reproduce.

--------------------- >> end captured logging << ---------------------
```

**Comment by Chancebair** [comment]
Created a PR to disable test: https://github.com/apache/incubator-mxnet/pull/13057

**Comment by mseth10** [comment]
I investigated this issue. The dropout symbol implementation calls a function BernoulliGenerate when MKL and OpenMP flags are on. This is where the flakiness comes from.

BernoulliGenerate uses multithreading and calls MKL library function viRngBernoulli to populate a mask vector. https://github.com/apache/incubator-mxnet/blob/master/src/operator/nn/dropout-inl.h#L82

When **p**=1.0, it is supposed to populate the mask vector **r** of length **n** with all 1s. But for a few values of **seed** internally generated, mask vector has n-1 1s and a 0 (located at a different index for a different flaky seed), which causes the error.

Also, the error only occurs when multi-threading used. It is not reproduced when single thread used.

I suspect that viRngBernoulli is not thread-safe.

@TaoLv @pengzhao-intel @ZhennanQin @xinyu-intel Can you please take a look? Would really like your inputs on the same. Thanks!

**Comment by pengzhao-intel** [comment]
@mseth10 thanks for the deep dive. I will look into the case and back to you soon.

**Comment by mseth10** [comment]
@pengzhao-intel Apparently, the issue is not with viRngBernoulli, it is with the streams being generated. In the present code, different seed is passed to vslNewStream for different threads. Making the seeds constant fixes the flaky issue. Please review. Fix in #13276 .
